File: LinkedList.py
# Linked List code

class Node:
    def __init__(self,data):
        self.data = data
        self.next = None


class LinkedList:
    
    def __init__(self):
        self.head = None
    
    
    
    def insert_at_end(self,data):
        
        if(self.head == None):
            node = Node(data)
            self.head = node
        
        else:
            
            temp = self.head
            while(temp.next):
                temp = temp.next
            
            temp.next = Node(data)
            
            
        
    
    
    # def delete(self,data):
        
    
    
    def printlinkedlist(self):
        
        temp = self.head
        while(temp):
            print(temp.data)
            temp = temp.next
        
        
        
ll = LinkedList()  
ll.insert(10)
ll.insert(12)
ll.insert(14)
ll.printlinkedlist()
    
    
    

File: LinkedList.cpp

#include <bits/stdc++.h>
using namespace std;


class Node {
    public:
        int data;
        Node* next;
        
        Node(int val)
        : data(val),
          next(NULL)
        {
            
        }
    
        
};

class LinkedList {
    public:
        Node* head;
        
        LinkedList()
            : head(NULL)
        {
            
        }
        
        
        ~LinkedList()
        {
            Node* tmp = head;
            while(tmp) {
                Node* nxt = tmp->next;
                delete tmp;
                tmp = nxt;
            }
            
        }
    
        void addNode(int val)
        {
            
            if(head == NULL) {
                Node* node = new Node(val);
                head = node;
                
            } else {
                Node* tmp = head;
                
                while(tmp->next) {
                    tmp = tmp->next;
                }
                
                Node* node = new Node(val);
                tmp->next = node;
                
            }
            
            
        }
     
    
        void printLinkedList() 
        {
            Node* tmp = head;
            while(tmp) {
                if(tmp->next == NULL){
                    cout<<tmp->data;
                }  
                else{
                    cout<<tmp->data<<"->" ;
                    
                }
                
                tmp = tmp->next;
             
            }
        }
    

};


int main() {
    
    LinkedList ll;
    ll.addNode(10);
    ll.addNode(20);
    ll.addNode(30);
    ll.addNode(40);
    ll.addNode(50);
    
    ll.printLinkedList();
	return 0;
}
